src/global/testStringUtil.cc

```diff
@@ -0,0 +1,110 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// System headers
+#include <filesystem>
+#include <fstream>
+#include <string>
+#include <random>
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "global/stringUtil.h"
+
+// Boost unit test header
+#define BOOST_TEST_MODULE stringUtil
+#include <boost/test/unit_test.hpp>
+
+namespace fs = std::filesystem;
+
+namespace {
+
+// Generate a random alphanumeric suffix for temp dirs
+static std::string random_suffix(size_t len = 8) {
+    static const char charset[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+    static thread_local std::mt19937 rng(std::random_device{}());
+    static thread_local std::uniform_int_distribution<size_t> dist(0, sizeof(charset) - 2);
+    std::string s;
+    s.reserve(len);
+    for (size_t i = 0; i < len; ++i) s.push_back(charset[dist(rng)]);
+    return s;
+}
+
+// RAII class to manage a temporary directory
+struct TempDir {
+    fs::path path;
+    explicit TempDir(std::string_view prefix = "global-test-") {
+        path = fs::temp_directory_path() / fs::path(prefix) / random_suffix();
+        fs::create_directories(path);
+    }
+    ~TempDir() {
+        std::error_code ec;
+        fs::remove_all(path, ec);
+    }
+};
+
+// Write data to a file, creating parent directories as needed
+static void write_file(const fs::path& p, std::string_view data) {
+    fs::create_directories(p.parent_path());
```

Comment:
The method may throw exceptions, unless you use the overloaded version of the function:
```
bool create_directories(const std::filesystem::path& p, std::error_code& ec );
```
Then you may check for the code:
```
BOOST_REQUIRE_MESSAGE(ec, "Couldn't create the directories: " << p);
```
This will call `operator bool()`, which returns `false` in case of any error. 

See:
- https://en.cppreference.com/w/cpp/filesystem/create_directory.html
- https://en.cppreference.com/w/cpp/error/error_code/operator_bool.html